.github/workflows/ci-cd-workflow.yml

```diff
@@ -40,19 +37,20 @@ jobs:
       - name: Build and Test
         run: mvn clean install
 
-#  deploy:
-#    name: Deploy to App Engine
-#    needs: build
-#    runs-on: ubuntu-latest
-#    environment: 'google_cloud'
-#    steps:
-#      - name: Checkout
-#        uses: actions/checkout@v2
-#      - name: Google App Engine
-#        uses: google-github-actions/deploy-appengine@v0.4.0
-#        with:
-#          project_id: bytecoders-coms4156
-#          version: main
-#          credentials: ${{ secrets.PHARMAID_GOOGLE_APPLICATION_CREDENTIALS }}
-#          flags: --no-cache
-#          deliverables: app.yaml
\ No newline at end of file
+  deploy:
+    runs-on: ubuntu-latest
+    needs: build
+    if: github.ref == 'refs/heads/main'
```

Comment:
I've not seen it before and usually it's a separate GitHub action, but let's give it a shot